modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -46,6 +48,8 @@ type FileServer struct {
 	Hide       []string `json:"hide,omitempty"`
 	IndexNames []string `json:"index_names,omitempty"`
 	Browse     *Browse  `json:"browse,omitempty"`
+
+	bufPools map[string]*sync.Pool
```

Comment:
This should actually be a global variable, we don't want to be making new pools just because configuration has changed.

And I think we can simply make one pool, unless you suspect the JSON and HTML outputs will have very different sizes (I doubt they will be very different -- at least, within 1 order of magnitude; i.e. JSON wouldn't be KBs while HTML is MBs I think).

Reply:
Yap, I took this from your `Encode` example.

> But if you think otherwise and/or have evidence that different pools perform better, I could be convinced.

I guess this depends on how widely the pool sizes differ from each other. If you think that the JSON and HTML outputs will have similar sizes, a global buffer would be perfectly fine for me.

Reply:
Okay, in that case let's just use one global pool like here: https://sourcegraph.com/github.com/caddyserver/caddy@c8b0a97b1c2524f7486aef38b2cf78a93d7b98b0/-/blob/modules/caddyhttp/templates/tplcontext.go#L275-279